src/Microsoft.Restier.Core/Api.cs

```diff
@@ -78,16 +78,40 @@ public static class Api // TODO GitHubIssue#25,#26 : transactions, exception fil
         {
             Ensure.NotNull(context, "context");
             var config = context.Configuration;
-            if (config.Model == null)
+            if (config.Model != null)
             {
-                var builder = context.GetApiService<IModelBuilder>();
-                if (builder != null)
-                {
-                    config.Model = await builder.GetModelAsync(new InvocationContext(context), cancellationToken);
-                }
+                return config.Model;
+            }
+
+            var builder = context.GetApiService<IModelBuilder>();
+            if (builder == null)
+            {
+                return null;
```

Comment:
Maybe we should throw if there is no model builder? It is more meaningful than a later null-reference exception.